File: backend/src/main/java/com/contractly/template/model/Template.java
package com.contractly.template.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Template {
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private String content;       // TipTap JSON / HTML
    private String variables;     // JSON array: ["client_name", "amount", "start_date"]
    private boolean isPublic;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: backend/src/models/Template.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface ITemplate extends Document {
  userId: mongoose.Types.ObjectId;
  title: string;
  description?: string;
  content?: string;
  variables?: any; // JSON array or object
  isPublic: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const TemplateSchema: Schema = new Schema(
  {
    userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    title: { type: String, required: true },
    description: { type: String },
    content: { type: String },
    variables: { type: Schema.Types.Mixed },
    isPublic: { type: Boolean, default: false },
  },
  { timestamps: true }
);

export default mongoose.model<ITemplate>('Template', TemplateSchema);
